src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBlob.java

```diff
@@ -143,6 +143,11 @@ private void checkClosed() throws SQLServerException {
     @Override
     public InputStream getBinaryStream() throws SQLException {
         checkClosed();
+        // If the LOB is currently streaming and the stream hasn't been read, read it.
+        if (!delayLoadingLob && null == value && !activeStreams.isEmpty()) {
```

Comment:
null check for activeStream

Reply:
I think this is fine, it's impossible for that variable to be null, and many other parts of this class doesn't null check that variable.

Reply:
it would be null if free is called before. That would be an error, but should not cause driver to crash on null exception

Reply:
Well...after free() the blob object is invalid, so the user shouldn't be using the blob after free() is called. Also, you'll see that on line 151 that this method wasn't doing a null check before anyways, but I'll leave it up to @rene-ye to make the call.

Reply:
After the lob is freed it won't get past checkClosed().